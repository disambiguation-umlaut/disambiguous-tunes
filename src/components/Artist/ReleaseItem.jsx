/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';


const ReleaseItem = ({ release }) => {

  
  return (
    <div>
      <h2>{release.title}</h2>
      {
        release.cover ? <img src={`http://coverartarchive.org/release/${release.id}/front`} alt={release.title} />
          : <p>no cover available</p>
      }
      <p>{release.date}</p>
    </div>
  );
};

ReleaseItem.propTypes = {
  release: PropTypes.arrayOf(
    PropTypes.shape(
      {
        title: PropTypes.string.isRequired, 
        id: PropTypes.string.isRequired,
        cover: PropTypes.boolean.isRequired,
        date: PropTypes.string.isRequired
      }
    )).isRequired
};


export default ReleaseItem;
