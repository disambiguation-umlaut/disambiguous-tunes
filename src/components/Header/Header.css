@import url('https://fonts.googleapis.com/css2?family=Russo+One&display=swap');

.header {
    margin: 0;
    text-align: center;
    height: 150px;
    background-color: darkslategray;
    border: solid 1px darkslategray;
}

h1 {
    margin-top: 1%;
    font-family: Russo One;
    font-size: 50pt;
    letter-spacing: 1pt;
    color: gainsboro;
    text-shadow: 0px 0px 20px darkcyan;
}