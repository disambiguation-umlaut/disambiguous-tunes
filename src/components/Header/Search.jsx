import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useHistory } from 'react-router-dom';

const Search = () => {
  const [search, setSearch] = useState('');
  const history = useHistory();
  
  console.log(search);

  // history.push('?search=whatever')

  const handleSubmit = e => {
    e.preventDefault();

    history.push(`/?search=${search}`);
  };

  return (
    <div>
      <form onSubmit={handleSubmit}>
        <input 
          type="text" 
          onChange = {(e) => setSearch(e.target.value)}
          value = {search}
          placeholder="Search..." 
        />
        <button>Search</button>
      </form>
    </div>
  );
};

Search.propTypes = {
  handleClick: PropTypes.func
};

export default Search;
