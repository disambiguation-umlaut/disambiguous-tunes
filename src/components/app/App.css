@import url('https://fonts.googleapis.com/css2?family=Russo+One&display=swap');

@import url('https://fonts.googleapis.com/css2?family=Spartan:wght@400;600&display=swap');

html, body {
    margin: 0;
    background-color: whitesmoke;
}