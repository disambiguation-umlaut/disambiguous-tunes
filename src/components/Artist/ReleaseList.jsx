import React from 'react';
import PropTypes from 'prop-types';
import ReleaseItem from './ReleaseItem';
import { Link } from 'react-router-dom ';

const ReleaseList = ({ releases }, { artist }) => {
  const releaseElements = releases.map(release => {
    <div key={release.id}>
      <Link to={`/release/${release.id}`}>
        <ReleaseItem
          release={release}
          artist={artist} 
        />
      </Link>
    </div>;
  });

  return (
    <div data-testid="releases">
      {releaseElements}
    </div>
  );
};

ReleaseList.propTypes = {
  releases: PropTypes.arrayOf(
    PropTypes.shape(
      {
        id: PropTypes.string.isRequired, 
        title: PropTypes.string.isRequired,
        cover: PropTypes.boolean.isRequired,
        date: PropTypes.string.isRequired
      }
    )).isRequired
};

export default ReleaseList;

