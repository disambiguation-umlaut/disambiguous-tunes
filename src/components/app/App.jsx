import React from 'react';
import { BrowerRouter as Router, Route, Switch } from 'react-router-dom';
import LandingPage from '../../container/LandingPage';
import Header from '../Header/Header';
import ArtistPage from '../../container/ArtistPage';
import ReleasePage from '../../container/ReleasePage';
import LyricsPage from '../../container/LyricsPage';

export default function App() {
  return (
    <>
      <Router>
        <Header />
        <Switch>
          <Route exact path="/" component={LandingPage} />
          <Route path="/artist/:id" component={ArtistPage} />
          <Route path="/release/:id" component={ReleasePage} />
          <Route path="/lyrics/:artist/:title" component={LyricsPage} />
        </Switch>
      </Router>
    </>
  );
}
