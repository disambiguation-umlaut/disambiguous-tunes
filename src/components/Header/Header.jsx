import React from 'react';
import Search from './Search';
import { Link } from 'react-router-dom';

const Header = () => {

    
  return (
    <div>
      <h1>DISAMBIGUOUS TUNES</h1>
      <Link to="/"><span>Back to Home</span></Link>
      <Search />
    </div>
  );
};

export default Header;
