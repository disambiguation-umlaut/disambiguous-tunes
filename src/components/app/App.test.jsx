import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import blueReleaseApi from '../../fixtures/blueReleaseApi.json';
import App from './App';
import Header from '../Header/Header';
import ReleaseItem from '../Artist/ReleaseItem';
import ReleaseList from '../Artist/ReleaseList';

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders Header', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders ReleaseList', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <ReleaseList 
          releases={blueReleaseApi.releases} 
          artist={'Blue Blue Blue'}/>
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

});
