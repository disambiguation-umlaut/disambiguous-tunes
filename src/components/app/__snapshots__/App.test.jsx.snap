// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <div>
    <h1>
      DISAMBIGUOUS TUNES
    </h1>
    <a
      href="/"
    >
      <span>
        Back to Home
      </span>
    </a>
    <div>
      <form>
        <input
          placeholder="Search..."
          type="text"
          value=""
        />
        <button>
          Search
        </button>
      </form>
    </div>
  </div>
  <div>
    <div
      data-testid="artists"
    />
  </div>
</DocumentFragment>
`;

exports[`App component renders Header 1`] = `
<DocumentFragment>
  <div>
    <h1>
      DISAMBIGUOUS TUNES
    </h1>
    <a
      href="/"
    >
      <span>
        Back to Home
      </span>
    </a>
    <div>
      <form>
        <input
          placeholder="Search..."
          type="text"
          value=""
        />
        <button>
          Search
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`App component renders ReleaseList 1`] = `
<DocumentFragment>
  <div
    data-testid="releases"
  >
    <div>
      <a
        href="/release/Blue Blue Blue/12503eef-7fca-4322-9012-2057995aaba7"
      >
        <div>
          <h2>
            Another Night Time Flight
          </h2>
          <p>
            no cover available
          </p>
          <p>
            2009-06-28
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        href="/release/Blue Blue Blue/1904a7ad-ca50-44f5-9b59-a6ed6d43c93c"
      >
        <div>
          <h2>
            Cookie In A Jar
          </h2>
          <p>
            no cover available
          </p>
          <p>
            1975
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        href="/release/Blue Blue Blue/1ef2a5c7-6fae-4857-8408-4f8f2606a8e4"
      >
        <div>
          <h2>
            Blue
          </h2>
          <p>
            no cover available
          </p>
          <p>
            1973
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        href="/release/Blue Blue Blue/2424deff-a37c-45c7-a91a-02e698b06ac8"
      >
        <div>
          <h2>
            Fools' Party
          </h2>
          <p>
            no cover available
          </p>
          <p />
        </div>
      </a>
    </div>
    <div>
      <a
        href="/release/Blue Blue Blue/2d4665e6-b974-42fe-b68e-1a3e3648a44c"
      >
        <div>
          <h2>
            Lonesome
          </h2>
          <p>
            no cover available
          </p>
          <p>
            1974
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        href="/release/Blue Blue Blue/4674b1b7-7713-4d6f-a801-5cfc059a699d"
      >
        <div>
          <h2>
            Women
          </h2>
          <p>
            no cover available
          </p>
          <p>
            1977-11-18
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        href="/release/Blue Blue Blue/50e56a40-6ae1-4a30-829a-9e18254b0c63"
      >
        <div>
          <h2>
            Danger Sign
          </h2>
          <p>
            no cover available
          </p>
          <p>
            1979
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        href="/release/Blue Blue Blue/5384a3aa-a636-4ecf-a4a1-549c95cbf86e"
      >
        <div>
          <h2>
            Another Night Time Flight / I'm Alone
          </h2>
          <p>
            no cover available
          </p>
          <p>
            1977
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        href="/release/Blue Blue Blue/65779fd8-4020-44dd-af98-3265383e1f65"
      >
        <div>
          <h2>
            Bring Back The Love
          </h2>
          <p>
            no cover available
          </p>
          <p>
            1977
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        href="/release/Blue Blue Blue/685ebc53-4d42-44a9-921d-fd1692b5e650"
      >
        <div>
          <h2>
            The L.A Sessions 1979-1982
          </h2>
          <p>
            no cover available
          </p>
          <p />
        </div>
      </a>
    </div>
    <div>
      <a
        href="/release/Blue Blue Blue/6f22fb54-e79a-4169-9c99-3b0c71873a6f"
      >
        <div>
          <h2>
            Love Sings
          </h2>
          <p>
            no cover available
          </p>
          <p>
            1979
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        href="/release/Blue Blue Blue/7090dd71-99d7-4518-a388-f553d510faa5"
      >
        <div>
          <h2>
            Stranger's Town
          </h2>
          <p>
            no cover available
          </p>
          <p>
            1979
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        href="/release/Blue Blue Blue/79e52dd3-179e-48c0-9403-072a5e8afee0"
      >
        <div>
          <h2>
            Gonna Capture Your Heart
          </h2>
          <p>
            no cover available
          </p>
          <p>
            1977
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        href="/release/Blue Blue Blue/931b833d-fa37-479e-a4c4-637420d997c0"
      >
        <div>
          <h2>
            Life in the Navy
          </h2>
          <p>
            no cover available
          </p>
          <p>
            1974
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        href="/release/Blue Blue Blue/a4c2c954-b6c6-46b6-8965-646fac7da85e"
      >
        <div>
          <h2>
            Another Night Time Flight
          </h2>
          <p>
            no cover available
          </p>
          <p>
            1977
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        href="/release/Blue Blue Blue/a5910dd8-0f65-461e-a3cf-8876fe4114de"
      >
        <div>
          <h2>
            I Wanna Go To New York
          </h2>
          <p>
            no cover available
          </p>
          <p>
            1984
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        href="/release/Blue Blue Blue/ba9e5911-456d-4c18-aafa-57bcc807767f"
      >
        <div>
          <h2>
            Red Light Song
          </h2>
          <p>
            no cover available
          </p>
          <p>
            1973
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        href="/release/Blue Blue Blue/d67af23c-d8ec-4081-ab16-23e29d91c5cd"
      >
        <div>
          <h2>
            Round And Round
          </h2>
          <p>
            no cover available
          </p>
          <p>
            1975
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        href="/release/Blue Blue Blue/e6e90a5d-614b-472e-b284-1dcd68824af3"
      >
        <div>
          <h2>
            Country Blue
          </h2>
          <p>
            no cover available
          </p>
          <p />
        </div>
      </a>
    </div>
    <div>
      <a
        href="/release/Blue Blue Blue/f15c3cde-a262-4cf3-b3cc-7f865768cf22"
      >
        <div>
          <h2>
            Blue
          </h2>
          <p>
            no cover available
          </p>
          <p>
            2002
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        href="/release/Blue Blue Blue/f4b824dc-75a5-40ef-ae74-dde78b68aa46"
      >
        <div>
          <h2>
            Don't Wanna Make You Cry / Moonlight
          </h2>
          <p>
            no cover available
          </p>
          <p>
            1982
          </p>
        </div>
      </a>
    </div>
    <div>
      <a
        href="/release/Blue Blue Blue/f66c7755-ec0a-4c22-bcd6-4e795797d3d5"
      >
        <div>
          <h2>
            Blue
          </h2>
          <p>
            no cover available
          </p>
          <p />
        </div>
      </a>
    </div>
    <div>
      <a
        href="/release/Blue Blue Blue/f8971ae7-bc9e-42df-9b31-529b9a92d99a"
      >
        <div>
          <h2>
            Life in the Navy
          </h2>
          <p>
            no cover available
          </p>
          <p>
            2002
          </p>
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
